pkg/server/handler.go

```diff
@@ -194,6 +196,7 @@ func (ctrl *Controller) decodeGithubCallbackResponse(resp *http.Response) (strin
 	var userProfile callbackResponse
 	err := json.NewDecoder(resp.Body).Decode(&userProfile)
 	if err != nil {
+		ctrl.log.WithError(err).Error("Decoding Github Callback Response failed")
 		return "", err
```

Comment:
The error is logged later in `callbackRedirectHandler`, no need to log it twice. Same for `decodeGitLabCallbackResponse` and `decodeGoogleCallbackResponse`.